onnxruntime/test/onnx/main.cc

```diff
@@ -257,12 +257,15 @@ int real_main(int argc, char* argv[], Ort::Env& env) {
     getchar();
   }
 
-  try {
```

Comment:
Is it possible to avoid changing the unit tests? We don't care their binary size.

Reply:
This is not currently enabled in the build script, when we enable -fno-exceptions, that will be global, this will cause warnings and possibly leaks, not sure if there is a solution to this

Reply:
the build will break when -fno-exceptions is specified if we don't. and to get the significant binary size reduction you have to build with that flag and not just disable the usage of exceptions in ORT.

Reply:
Actually I think we need a different approach here. If exceptions are disabled, any test that throws has to be disabled given that will result in an abort. So instead of using ORT_TRY/ORT_CATCH, I think we need to #ifdef out the entire test.

---
In reply to: [475891852](https://github.com/microsoft/onnxruntime/pull/4894#discussion_r475891852) [](ancestors = 475891852)